### PR TITLE
chore: add deprecated flags to api methods

### PR DESCRIPTION
**Pull request by kuruk-mm**
Extract from https://github.com/decentraland/js-sdk-toolchain/pull/743

### CODE REVIEW

File: scripts/rpc-api-generation/index.ts [hunk 1]
```diff
@@ -94,7 +108,7 @@ async function internalCompile() {
 `
     apisDTsContent += readFileSync(moduleDTsPath).toString()
 
-    rmSync(apiModuleDirPath, { recursive: true, force: true })
+    // rmSync(apiModuleDirPath, { recursive: true, force: true })
```
Comment (leanmendoza) [inline scripts/rpc-api-generation/index.ts:111]:
```suggestion
    rmSync(apiModuleDirPath, { recursive: true, force: true })
```

File: scripts/rpc-api-generation/index.ts [hunk 2]
```diff
@@ -54,9 +65,12 @@ async function internalCompile() {
     if (NON_EXPOSED_LIST_NAMES.includes(api.name)) continue
     const types: Set<string> = new Set()
     const functions: string[] = []
+    const deprecatedMethods = deprecatedApiMethods[api.name] || []
+
     for (const [methodName, method] of Object.entries(api.def.methods)) {
       types.add(method.requestType)
       types.add(method.responseType)
+
```
Comment (leanmendoza) [inline scripts/rpc-api-generation/index.ts:73]:
```suggestion
```

File: scripts/rpc-api-generation/index.ts [hunk 3]
```diff
@@ -22,6 +22,14 @@ type ProtoServiceDefinition = {
   >
 }
 
+const deprecatedApiMethods: Record<string, string[]> = {
+  CommunicationsController: ['send'],
+  EngineApi: ['sendBatch', 'subscribe', 'unsubscribe'],
+  EthereumController: ['requirePayment', 'signMessage', 'convertMessageToObject', 'getUserAccount'],
+  Players: ['getPlayerData', 'getPlayersInScene', 'getConnectedPlayers'],
```
Comment (pbosio) [inline scripts/rpc-api-generation/index.ts:29]: should we add some text with the deprecated tag? i.e: "@deprecated use <this other api> instead"?

File: scripts/rpc-api-generation/index.ts [hunk 4]
```diff
@@ -22,6 +22,14 @@ type ProtoServiceDefinition = {
   >
 }
 
+const deprecatedApiMethods: Record<string, string[]> = {
```
Comment (gonpombo8) [inline scripts/rpc-api-generation/index.ts:25]: wait, why are we duplicating all the depecreated methods here ?
we already have this info on the .proto file. We can do some regex to find all the methods that are depcreated.
This way we have only one source of truth

**Review by gonpombo8 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**